composer.json

```diff
@@ -15,7 +15,7 @@
         "symplify/phpstan-extensions": "^10.0",
         "symplify/easy-coding-standard": "^10.0",
         "symplify/rule-doc-generator": "^10.0",
-        "rector/rector-src": "dev-main#6ee4324",
+        "rector/rector-src": "dev-main",
```

Comment:
This was the root of your problem with missing services.
I was probably testing some PR on Rector and forgot to release it back here. My bad.

After the version change, the extension have to be updated with:

```bash
composer install
```

This invokes the composer pugin and registeres services from Rector packages.

Reply:
Aha, it's in Rector run itself, not in the tests :) 

Hm, interesting.

Reply:
Locally it seems ok, so it's probably some glitch in Github Action... I'll look into it

Reply:
Found it :) the same way we include local `config/config.php` in tests, it must be included in the `rector.php` too.

Otherwise all but local services are loaded. That's why tests passes, because the config is always loaded there:
https://github.com/rectorphp/rector-doctrine/blob/07a8a64857c82c511c3aa18636ad44e784636f15/tests/Set/DoctrineORM29Set/config/php81_nested_attributes.php#L11